java/src/jmri/jmrit/beantable/signalmast/SignalMastTableDataModel.java

```diff
@@ -355,6 +382,391 @@ protected boolean matchPropertyName(java.beans.PropertyChangeEvent e) {
         return super.matchPropertyName(e);
     }
 
+    /**
```

Comment:
Thanks for your detailed review @rhwood. At the time the Aspect combos stopped working, the most recent change was the introduction of TableRowSorter. What that did solve (when properly converting row index from Table to Model), is the mix up of comboboxes appearing in the wrong place after changing the order of columns.
![456ish mast pairs error](https://cloud.githubusercontent.com/assets/14811327/20590555/1aa18af0-b224-11e6-87f7-03e4a9e893b5.png)
So that's where I started. #2559 was not visible to me while trying to get this to work and seems pure genius, but coming back to RowSorter wasn't something I considered.
To create custom (per line) comboboxes without subclassing the BeanTable I found no simpler way than this, elegantly using table.setDefaultEditor() in the available configValueColumn(). We do need such a thing for other applicatios, like the Signaling Mast Pairs panel (Mast Logic > Edit).
Apart from small stuff like UI fixes etc. (which we could cherry pick later).